vault/resource_aws_auth_backend_config_identity_test.go

```diff
@@ -0,0 +1,163 @@
+package vault
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestAccAwsAuthBackendConfigIdentity(t *testing.T) {
+	backend := acctest.RandomWithPrefix("aws")
+	resourceName := "vault_aws_auth_backend_config_identity.config"
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		Providers:    testProviders,
+		CheckDestroy: testAccCheckAwsAuthBackendConfigIdentityDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsAuthBackendConfigIdentity_basic(backend),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
looks like we can also replace some of these with constants. we can leave the list fields as unchanged though